### ISSUE TITLE
About Grad

### ISSUE THREAD

**Issue by Chantec** [ChrisDud0257/SSL#4]
Hello, I have recently been using L1loss(HR, SR), but I found that during sampling, each sampling step requires about 14GB of GPU memory  if I don't comment out with torch.no_grad. It results in an out-of-memory (OOM) error. If I only leave one sample for training, the results are very bad. How do you handle this situation?
![image](https://github.com/user-attachments/assets/f33b0a04-afb6-4bc3-8a94-913bd429d983)

**Comment by ChrisDud0257** [comment]
Two ways to deal with it:
1.In training stage, just need one-step denoise by this formula:
![image](https://github.com/user-attachments/assets/db43fd61-2632-4e4a-a6e0-8d138469cc49)
The results will not be bad in the inference stage.

2.Decrease the denoise step from 1000 to 20 or less. Change other scheduler to complete this progress. In [PASD](https://github.com/yangxy/PASD),  UniPC scheduler could obtain good results within 20 steps.

**Comment by Chantec** [comment]
> Two ways to deal with it: 1.In training stage, just need one-step denoise by this formula: ![image](https://private-user-images.githubusercontent.com/128046921/374892286-db43fd61-2632-4e4a-a6e0-8d138469cc49.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mjg0Njc0NTIsIm5iZiI6MTcyODQ2NzE1MiwicGF0aCI6Ii8xMjgwNDY5MjEvMzc0ODkyMjg2LWRiNDNmZDYxLTI2MzItNGU0YS1hNmUwLThkMTM4NDY5Y2M0OS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQxMDA5JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MTAwOVQwOTQ1NTJaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1iZjFiYzAwOWI2NTZjYTBjZTVhOWJiMjliNDBlMDYzNTdlNTcyMzhhYmY3YjU0NDBjOTRiYWQ5YmI5N2FhOWUwJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.OYJnmqgkuSIHTYoBIC5Vi2dXFbSPjJY1RwVK2fODItQ) The results will not be bad in the inference stage.
> 
> 2.Decrease the denoise step from 1000 to 20 or less. Change other scheduler to complete this progress. In [PASD](https://github.com/yangxy/PASD), UniPC scheduler could obtain good results within 20 steps.

In my previous idea, when t=0, the predicted x_0 can be decoded to img, and then loss can be obtained. You mean that x_0 can be obtained for any step that time  in [T,0], and then decode it into an img and perform loss?
